src/test/java/finrem/HighLevelDataSetupApp.java

```diff
@@ -1,39 +1,114 @@
+        super(dataSetupEnvironment, DEFINITIONS_PATH);
     }
 
     public static void main(String[] args) throws Throwable {
         main(HighLevelDataSetupApp.class, args);
     }
 
     @Override
-    protected void doLoadTestData() {
-        List<String> definitionFileResources = getAllDefinitionFilesToLoadAt(definitionsPath);
-        CcdEnvironment currentEnv = (CcdEnvironment) getDataSetupEnvironment();
+    public void addCcdRoles() {
+        if (getDataSetupEnvironment() == CcdEnvironment.PREVIEW) {
+            logger.info("Adding CCD roles");
+            CCD_ROLES.forEach(this::addCcdRole);
+        }
+    }
+
+    @Override
+    public void importDefinitions() {
+        logger.info("Importing CCD definitions");
+
+        CcdEnvironment ccdEnvironment = (CcdEnvironment) getDataSetupEnvironment();
+        logger.info("CCD environment: {}", ccdEnvironment);
         try {
-            if (currentEnv != null && !SKIPPED_ENVS.contains(currentEnv)) {
-                importDefinitions();
+            if (ccdEnvironment != null && !SKIPPED_ENVS.contains(ccdEnvironment)) {
+                importDefinitionsAt(DEFINITIONS_PATH);
             } else {
-                definitionFileResources.forEach(file ->
-                        System.out.println("definition file \"" + file + "\" is skipped on " + currentEnv));
+                logger.info("CCD definitions import skipped for {} environment", ccdEnvironment);
             }
         } catch (Exception e) {
-            System.out.println("Error on uploading ccd definition file - " + e.getMessage());
-            // exit the process to fail jenkin pipeline
+            logger.error("Error when importing CCD definitions", e);
+            // Exit the process to fail Jenkins pipeline
             System.exit(1);
         }
     }
+
+    @Override
+    public void createRoleAssignments() {
+        // No implementation required
+    }
+
+    @Override
+    protected boolean shouldTolerateDataSetupFailure(Throwable e) {
+        if (getDataSetupEnvironment() == CcdEnvironment.PREVIEW) {
+            logger.error("Data Setup failure ignored", e);
+            return true;
+        } else {
+            return super.shouldTolerateDataSetupFailure(e);
+        }
+    }
+
+    /**
+     * Workaround for <a href="https://tools.hmcts.net/jira/browse/CCD-5362">CCD-5362</a>.
+     * @param definitionsPath definitions path
+     */
+    @Override
+    protected void importDefinitionsAt(String definitionsPath) {
+        List<String> definitionFileResources = getAllDefinitionFilesToLoadAt(definitionsPath);
+        logger.info("{} definition files will be uploaded to '{}' on {}.", definitionFileResources.size(),
+                BeftaMain.getConfig().getDefinitionStoreUrl(), getDataSetupEnvironment());
+        String message = "Couldn't import {} - Exception: {}.\n\n";
+        try {
+            for (String fileName : definitionFileResources) {
+                try {
```

Comment:
Should we move this second try block into a separate method? We have nested try catch block so perhaps we can move this out into a method like `importSingleDefinition(String fileName)` and then move the `message` variable into that method too 🤔

Reply:
This code has been copied from https://github.com/hmcts/befta-fw/blob/b030e3e954bcb5920854022f8d3f8bb59f2ee9de/src/main/java/uk/gov/hmcts/befta/dse/ccd/DataLoaderToDefinitionStore.java#L328

Essentially I want to do exactly the same as the parent class but not throw any RuntimeException in preview so ideally I want to minimise changes but I accept your comment (perhaps you could create a PR on their repo too 😉 )